### ISSUE TITLE
Insane packet loss

### ISSUE THREAD

**Issue by sandikata** [ArmynC/ArminC-AutoExec#12]
Insane packet loss (about 50 - 60%), for the protocol i am on fiber connection 500MBps DL/UP, it can't be issue, as any other games have no network issues

## Expected behavior

Packet loss to be in acceptable amount not higher than 5%

## Current behavior

Packet loss going insane after 5-10 minutes, at the begining it is about 5-10%, then it is going up to 50 - 60%

## Your environment

* Config version (date): Latest
* Launch options (if any):
* Game version (beta?):
* Operating system and version [OS] (tweaked?): Gentoo Linux
* Graphics card [GPU]: RX6750XT
* Processor [CPU]: Ryzen 7 5800x
* Storage [SSD/HDD]: NVME
* RAM size [MB/GB]: 64GB
* Anything else:

## Checklist

- [x] I searched the [current issues](https://github.com/ArmynC/ArminC-AutoExec/issues) and the issue is not known.
- [x] I searched on Google, Steam, YouTube, Reddit and there's no fix.
- [ ] I am mostly sure that the problem is related to the config and isn't my fault.

**Comment by armync** [comment]
That's very strange. If it hadn't been on a networking side, I could have attributed this to a process leak or some sort. You should check the behaviour through the debug script (see bind "[") and console debug (bind "]" followed by pressing "`" to open the console). In your case this can be due to multiple reasons. You must find what is the culprit. For this, try to remove the entire "cfg" folder of the game, alongside the autoexec and the rest of _defaulted_ configs and such. Check afterwards the behaviour using the default settings.

Let's suppose both cases (be it the game's problem or my config). You may try to use "cl_net_buffer_ticks" command. Right now, if you have the latest version, it is set to 0. Try to increase it to 1, or even to 2 if desired.

Also, "net_client_steamdatagram_enable_override" may be set to -1 (generally not recommended) if already forced to 2. Please check the reset of network.cfg cvars.

I can see that you are using Linux. During my gameplay on a distro such as openSUSE, on CS2 I could notice some stranger behaviours such an unstable game, though it was due to the Vulkan backend, or even nvidia drivers (now seems to be fixed), not necessarily something related to networking. I suppose that the game is played with X11, not Wayland (worth a shot though). It may be the case where Valve failed to optimize it enough.

If you run some sorts of dockers, servers or so on, please check your linux networking (settings) or processes. It may eat your bandwidth too.

Sometimes, by clearing the cache (though on Linux, as said above, should be on Vulkan - https://unix.stackexchange.com/questions/604737/where-does-mesa-store-its-vulkan-shader-cache-on-linux). Don't forget about the Steam's pre-shader cache. Toggle that too (off/on on case - generally is recommended to have it disabled due to no real usage anymore).

Other than that, you can try to debug it with tools such as:

https://packetlosstest.com/ (set desired game)
https://www.pingplotter.com/products/standard/ (you may use cs2 servers, community or official ones I guess) via Wine if compatible or tools such as maybe gnome networking, smokeping etc.

If the network is going through a router, please check your QoS, Game Accelerators, prioritizations for gaming side.

In the case of a global problem (not related just to my config), please try to redirect this issue ticket to https://github.com/ValveSoftware/csgo-osx-linux/issues as they may fix your problem, or at least debug it properly.

Looking up to your answer. Thanks!

**Comment by sandikata** [comment]
It is not related to your configuration, before armory update i was using your configs and they were fine.

What i found is that, limiting bandwidth to extremely restricted, it improves, but with some impact.

https://www.reddit.com/r/GlobalOffensive/comments/1g02b4y/packet_loss_in_cs2_only_since_latest_updates/

**Comment by sandikata** [comment]
I can see that you are using Linux. During my gameplay on a distro such as openSUSE, on CS2 I could notice some stranger behaviours such an unstable game, though it was due to the Vulkan backend, or even nvidia drivers (now seems to be fixed), not necessarily something related to networking. I suppose that the game is played with X11, not Wayland (worth a shot though). It may be the case where Valve failed to optimize it enough.


Actually i am forced to use wayland, as it feels much better.
It is always case where Valve do not care about Linux version of the game, so for csgo it took them like 10 years to optimize it, and then they released this crap.

Nvidia on linux is really bad last years. I have used RTX3050 for some time, and then switched back to AMD with RX6750T.

![изображение](https://github.com/user-attachments/assets/48fad30e-940f-49b3-996e-b9028c322675)

**Comment by sandikata** [comment]
If you run some sorts of dockers, servers or so on, please check your linux networking (settings) or processes. It may eat your bandwidth too.

Don't use any dockers.

Actually my conenction is stable at all. (It is just CS2 with this problem).

https://www.speedtest.net/result/16874268882.png Munich
https://www.speedtest.net/result/16874270472.png Bucurest
https://www.speedtest.net/result/16874273200.png Moscow
https://www.speedtest.net/result/16874275075.png Belgrade
https://www.speedtest.net/result/16874278074.png Sofia

**Comment by sandikata** [comment]
Sometimes, by clearing the cache (though on Linux, as said above, should be on Vulkan - https://unix.stackexchange.com/questions/604737/where-does-mesa-store-its-vulkan-shader-cache-on-linux). Don't forget about the Steam's pre-shader cache. Toggle that too (off/on on case - generally is recommended to have it disabled due to no real usage anymore).

I am not sure if it's not used at all anymore. Do you have any document which describe it.

**Comment by sandikata** [comment]
Any idea on.. Why is stuttering that bad, even if i have constant 200+ fps?

**Comment by armync** [comment]
Yeah, your connection should be ruled out ouf this as a non-problem. I think that the pre-shader is still used but with no real benefits today (I don't have an official source only than the topics on the communities).  This should be mostly related to CS2's implementation on Linux (which now is native not on proton)... Other than blindly testing the OS with different values for networking I'm not sure if you have any alternative. 

Put an output of your gaming experience (using the included hud debuger (key "[") and console debuger (key "]")) in the Valve's cs linux github repo (check the link above) and hopefully someone may take some notes... 

Oh! You may try emailing "cs2team@valvesoftware.com" with a subject such as **[CS2 Bug] Linux incoherent packet loss** and once again, insert all your findings, including the snippings from above and those of hud-debugging and console output script. Make sure you type the details (such as environment too) in a consistent way and make a request you don't make it too long or too difficult to understand. You may be lucky and someone will respond (it was my case). I can assure you they will read at least!

The game stuttering may be sadly related to the Linux build of CS2 :( I may try capping the maximal fps or once again, clearing on cache side. DXVK Async can't be an option anymore due to it being what I think a native vulkan build. Finally, a distro hopping _may_ solve the case. Right now Counter-Strike   cvars are severely limited to dev (compared to older games such as 1.6 or even team fortress 2 configs). Sadly, there aren't many cvars that may fix your situation.

**Comment by sandikata** [comment]
What i've found as workaround for packet loss is reducing Connection Bandwidth from Unlimited to Extremely Limited and no buffer, so in the worst case i got 5 - 6% loss (acceptable), but most of the time it is without loss. However, feels a little slower reaction. Another thing is that i am using Variable Refresh Rate (My monitor support FreeSync). I hope it will be improved (at least to the level before Armory update), it was butter smooth on Linux (with higher framerate than on Windows), in your opinion is it possible valve to brake Linux native game (just to make Windows one to look better)? I have such feeling... Maybe it was bad for Windows, and they made some changes which impact Linux performance.

**Comment by armync** [comment]
The rate reduction method is viable just under its lowest value?

![image](https://github.com/user-attachments/assets/16107ab3-302f-4c6b-8517-a6d1aeaf5957)

Anyway, it shouldn't be used for long as it indeed may restrict your gameplay experience. At least for the time being, you've found a workaround which is good.

By checking it up, the latest drivers and latest game updates made CS on Linux a lot better than half a year/one year or earlier. From what it seems, based on your reports, it may be a bug induced by Valve or something on your Linux distro. Valve right now is baking up Linux development on many levels (Deck, Proton, sponsorships), so there is no reason (on the contrary) for them to break the game. I'll recommend you to report on their e-mail as it may be produced by a faulty build under Armory. Maybe the next 1/2 updates should fix it, or I hope so.

**Comment by sandikata** [comment]
Current rate i am using to have acceptable experience is 98304.

Distro is unchanged at all, a lot of peoples (mostly Windows ones) have complained about the same.
We're waiting for new update to come. There's no other option.

**Comment by armync** [comment]
So I understand that only the lowest value (no mids) should theoretically fix the packet losses. (_posterity_). On the distro idea, I’m not necessarily saying that the settings were changed by you, though it’s good to know that you didn’t alter them, but sometimes even the given values can be modified by the team/volunteers as they may differ between one and other. 

I was unaware about an issue on all systems, that's very strange, this can be a regress after all 😞. Please keep us posted after the next update
